README.md

```diff
@@ -19,32 +21,41 @@ composer require fortuneglobe/types
 
 ### Available base types
 
-* StringType
-* FloatType
-* IntType
-* UuidType (extends StringType)
+* AbstractStringType
+* AbstractFloatType
+* AbstractIntType
+* AbstractUuidType (extends AbstractStringType)
 
 ### Possible exceptions
 
 ```
 \Fortuneglobe\Types\Exceptions\InvalidArgumentException
                                |- InvalidFloatValueException
                                |- InvalidIntValueException
-                               |- InvalidStringValueException
-                               `- InvalidFloatValueException
+                               `- InvalidUuid4ValueException
 ```
   
 
-### StringType
+### AbstractStringType
 
 ```php
 <?php declare(strict_types=1);
 
 namespace YourVendor\YourProject;
 
-use Fortuneglobe\Types\StringType;
+use Fortuneglobe\Types\AbstractStringType;
+use Fortuneglobe\Types\Exceptions\InvalidArgumentException;
 
-final class UserId extends StringType {}
+final class UserId extends AbstractStringType 
+{
+    protected function guardValueIsValid( string $value ) 
```

Comment:
The return type is omitted.

```php
protected function guardValueIsValid( string $value ): void
```

Regarding to this the guard methods in all abstract types need to be fixed.

Reply:
Fixed in 70fe97374eeed55e6dce6d5d1872fd2f9ec9974c